api/app/signals/apps/email_integrations/actions/assigned_action.py

```diff
@@ -0,0 +1,111 @@
+# SPDX-License-Identifier: MPL-2.0
+# Copyright (C) 2021 - 2022 Gemeente Amsterdam
+import logging
+from email.utils import formataddr
+
+from django.conf import settings
+from django.contrib.auth import get_user_model
+from django.core.mail import send_mail
+from django.template import Context, Template, loader
+
+from signals.apps.email_integrations.actions.abstract import AbstractSystemAction
+from signals.apps.email_integrations.exceptions import URLEncodedCharsFoundInText
+from signals.apps.email_integrations.models import EmailTemplate
+from signals.apps.email_integrations.utils import make_email_context
+from signals.apps.signals.models import Department, Signal
+
+logger = logging.getLogger(__name__)
+
+User = get_user_model()
+
+
+class AssignedAction(AbstractSystemAction):
+    """
+    This e-mail action is triggered when a signal is assigned to a
+    user or the users' department
+    """
+
+    key = EmailTemplate.SIGNAL_ASSIGNED
+    subject = """Melding {{ formatted_signal_id }} is toegewezen aan
+              {% if assigned_to_user %} jou{% else %} {{ assigned_to_department }}{% endif %}"""
+
+    note = None
+
+    def __call__(self, signal, dry_run=False, recipient=None, assigned_to=None):
+        if self.rule(signal):
+            if dry_run:
+                return True
+
+            if self.send_mail(signal, recipient=recipient, assigned_to=assigned_to):
+                self.add_note(signal)
+                return True
+
+        return False
+
+    def render_mail_data(self, context):
+        """
+        Renders the subject, text message body and html message body
+        """
+        try:
+            email_template = EmailTemplate.objects.get(key=self.key)
+
+            rendered_context = {
+                'subject': Template(email_template.title).render(Context(context)),
+                'body': Template(email_template.body).render(Context(context, autoescape=False))
+            }
+
+            subject = Template(email_template.title).render(Context(context, autoescape=False))
+            message = loader.get_template('email/_base.txt').render(rendered_context)
+            html_message = loader.get_template('email/_base.html').render(rendered_context)
+        except EmailTemplate.DoesNotExist:
+            logger.warning(f'EmailTemplate {self.key} does not exists')
+
+            subject = Template(self.subject).render(Context(context, autoescape=False))
+            message = loader.get_template('email/assigned_default.txt').render(context)
```

Comment:
While I like that you added a sane default for the mail template I don't think you should override this whole method to add it. Possible add a `get_template` method to the `AbstracAction` class that uses the default (fallback) templates unless you override it.

@vanbuiten what is your opinion here?

Reply:
For now went with the suggestions of @vanbuiten.